include/pisa/wand_utils.hpp

```diff
@@ -11,10 +11,13 @@ enum class partition_type { fixed_blocks, variable_blocks };
 template <typename Scorer = bm25>
 std::pair<std::vector<uint32_t>, std::vector<float>>
 static_block_partition(binary_freq_collection::sequence const &seq,
-                       std::vector<float> const &norm_lens) {
+                       std::vector<float> const &norm_lens,
+                       const uint64_t in_block_size = 0) {
```

Comment:
I think it would be more clear if we used `std::optional` here, explicitly indicating when a block size is defined and when it's not, instead of using a magic value 0:

```cpp
std::optional<uint64_t> const = std::nullopt) {
```

Reply:
I have decided it's better to not allow default parameters here. Instead, I have built the default constructor for either the `VariableBlock` or `FixedBlock` type to directly read the corresponding value from the config file. So, instantiating a default object will give us the default configuration, which can be used directly. Please check it out and let me know what you think - I believe this approach is much more robust and readable.